🐛 [Bug]: Middleware timeout returns nil by default and results in a panic.

### Bug Description

Hi, when I was running my application, all endpoints were panicking and crashing, and after I traced the panic error it came from the timeout middleware that I declared by default on the global middleware. 
I hope this will resolve the error that occurred. 

### How to Reproduce

when i do `curl` on my endpoint it will panic and cause crash. 

`root@root:~# curl localhost:8080/api/auth/login`

and displays output like this, then exit status 2.
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xac9823]

goroutine 18 [running]:
github.com/gofiber/fiber/v3/middleware/timeout.runHandler({0xdeffa0, 0x3d6008}, 0x0?, {0x0, 0x0, 0x0, 0x0})
	/path/to/go/pkg/mod/github.com/gofiber/fiber/v3@v3.0.0-beta.5/middleware/timeout/timeout.go:43 +0x43
main.InitServer.New.func2({0xdeffa0, 0x3d6008})
	/path/to/go/pkg/mod/github.com/gofiber/fiber/v3@v3.0.0-beta.5/middleware/timeout/timeout.go:22 +0xc5
github.com/gofiber/fiber/v3.(*DefaultCtx).Next(0x68808?)
	/path/to/go/pkg/mod/github.com/gofiber/fiber/v3@v3.0.0-beta.5/ctx.go:1171 +0x47
github.com/gofiber/fiber/v3/middleware/etag.New.func1({0xdeffa0, 0x3d6008})
	/path/to/go/pkg/mod/github.com/gofiber/fiber/v3@v3.0.0-beta.5/middleware/etag/etag.go:33 +0xd1
github.com/gofiber/fiber/v3.(*DefaultCtx).Next(0x98808?)
	/path/to/go/pkg/mod/github.com/gofiber/fiber/v3@v3.0.0-beta.5/ctx.go:1171 +0x47
main.InitServer.New.func1({0xdeffa0, 0x3d6008})
	/path/to/go/pkg/mod/github.com/gofiber/fiber/v3@v3.0.0-beta.5/middleware/requestid/requestid.go:40 +0x144
github.com/gofiber/fiber/v3.(*DefaultCtx).Next(0x988d8?)
	/path/to/go/pkg/mod/github.com/gofiber/fiber/v3@v3.0.0-beta.5/ctx.go:1171 +0x47
github.com/gofiber/fiber/v3/middleware/compress.New.func3({0xdeffa0, 0x3d6008})
	/path/to/go/pkg/mod/github.com/gofiber/fiber/v3@v3.0.0-beta.5/middleware/compress/compress.go:54 +0x50
github.com/gofiber/fiber/v3.(*DefaultCtx).Next(0x89182?)
	/path/to/go/pkg/mod/github.com/gofiber/fiber/v3@v3.0.0-beta.5/ctx.go:1171 +0x47
github.com/gofiber/fiber/v3/middleware/helmet.New.func1({0xdeffa0, 0x3d6008})
	... (stack trace continues)
```

And other accesses also experience the same thing, when I access the endpoint on the CLI and website, the result is still panic and crash. 

### Expected Behavior

I don't know whether this is an error from the fiber middleware library which doesn't indicate a fixed value so that it doesn't produce nil, or my own error in forgetting to configure it, and the panic happens on all endpoints because I declared this middleware globally and by default.

but for sure, I declared the middleware from the fiber library and used it by default like other middleware. 
Is it mandatory to modify the timeout middleware first? 

### Fiber Version

Fiber V3 latest

### Code Snippet (optional)

```go
package server

import (
some import...
)

type Server struct {
        app *fiber.App
}

func InitServer() *Server {
        s := &Server{
                app: fiber.New(fiber.Config{
                        AppName:       "Myapp",
                        StrictRouting: true,
                }),
        }

        // DB MIGRATE
        if r := db.DB.AutoMigrate(
                &models.User{},
                &models.Tickets{},
        ); r != nil {
                logger.Error.Fatalf("Database migrattion failed\nError: %v", r)
        }

        // global middleware
        s.app.Use(cors.New())
        s.app.Use(helmet.New())
        s.app.Use(requestid.New())

        s.app.Use(timeout.New(timeout.ConfigDefault.OnTimeout)) //I declared this as default on global, and all endpoints panicked.

        s.app.Use(recover.New(recover.Config{
                EnableStackTrace: true, // The panic is captured by the recover middleware on the output endpoint.
        }))

      // register routes
        routes.RegisterRoutes(s.app)

        return &Server{app: s.app}
}
```

### Checklist:

- [x] I agree to follow Fiber's [Code of Conduct](https://github.com/gofiber/fiber/blob/main/.github/CODE_OF_CONDUCT.md).
- [x] I have checked for existing issues that describe my problem prior to opening this one.
- [x] I understand that improperly formatted bug reports may be closed without explanation.

--- a/docs/middleware/timeout.md
+++ b/docs/middleware/timeout.md
@@ -4,15 +4,16 @@ id: timeout
 
 # Timeout
 
-There exist two distinct implementations of timeout middleware [Fiber](https://github.com/gofiber/fiber).
+The timeout middleware limits how long a handler can take to complete. When the
+deadline is exceeded, Fiber responds with `408 Request Timeout`. The middleware
+uses `context.WithTimeout` and makes the context available through
+`c.Context()` so that your code can listen for cancellation.
 
-## New
-
-As a `fiber.Handler` wrapper, it creates a context with `context.WithTimeout` which is then used with `c.Context()`.
-
-If the context passed executions (eg. DB ops, Http calls) takes longer than the given duration to return, the timeout error is set and forwarded to the centralized `ErrorHandler`.
-
-It does not cancel long running executions. Underlying executions must handle timeout by using `context.Context` parameter.
+:::caution
+`timeout.New` wraps your final handler and can't be added with `app.Use` or
+used in a middleware chain. Register it per route and avoid calling
+`c.Next()` inside the wrapped handler—doing so will panic.
+:::
 
 ## Signatures
 
@@ -22,45 +23,58 @@ func New(handler fiber.Handler, config ...timeout.Config) fiber.Handler
 
 ## Examples
 
-Import the middleware package that is part of the Fiber web framework
+### Basic example
+
+The following program times out any request that takes longer than two seconds.
+The handler simulates work with `sleepWithContext`, which stops when the
+request's context is canceled:
 
 ```go
+package main
+
 import (
+    "context"
+    "fmt"
+    "log"
+    "time"
+
     "github.com/gofiber/fiber/v3"
     "github.com/gofiber/fiber/v3/middleware/timeout"
 )
-```
-
-After you initiate your Fiber app, you can use the following possibilities:
-
-```go
-func main() {
-    app := fiber.New()
-    h := func(c fiber.Ctx) error {
-        sleepTime, _ := time.ParseDuration(c.Params("sleepTime") + "ms")
-        if err := sleepWithContext(c.Context(), sleepTime); err != nil {
-            return fmt.Errorf("%w: execution error", err)
-        }
-        return nil
-    }
-
-    app.Get("/foo/:sleepTime", timeout.New(h, timeout.Config{Timeout: 2 * time.Second}))
-    log.Fatal(app.Listen(":3000"))
-}
 
 func sleepWithContext(ctx context.Context, d time.Duration) error {
-    timer := time.NewTimer(d)
-
     select {
+    case <-time.After(d):
+        return nil
     case <-ctx.Done():
-        if !timer.Stop() {
-            <-timer.C
-        }
-        return context.DeadlineExceeded
-    case <-timer.C:
+        return ctx.Err()
     }
-    return nil
 }
+
+func main() {
+    app := fiber.New()
+
+    handler := func(c fiber.Ctx) error {
+        delay, _ := time.ParseDuration(c.Params("delay") + "ms")
+        if err := sleepWithContext(c.Context(), delay); err != nil {
+            return fmt.Errorf("%w: execution error", err)
+        }
+        return c.SendString("finished")
+    }
+
+    app.Get("/sleep/:delay", timeout.New(handler, timeout.Config{
+        Timeout: 2 * time.Second,
+    }))
+
+    log.Fatal(app.Listen(":3000"))
+}
+```
+
+Test it with curl:
+
+```bash
+curl -i http://localhost:3000/sleep/1000   # finishes within the timeout
+curl -i http://localhost:3000/sleep/3000   # returns 408 Request Timeout
 ```
 
 ## Config
@@ -72,19 +86,7 @@ func sleepWithContext(ctx context.Context, d time.Duration) error {
 | OnTimeout | `fiber.Handler`    | Handler executed when a timeout occurs. Defaults to returning `fiber.ErrRequestTimeout`. | `nil`  |
 | Errors    | `[]error`          | Custom errors treated as timeout errors.                            | `nil`  |
 
-Test http 200 with curl:
-
-```bash
-curl --location -I --request GET 'http://localhost:3000/foo/1000' 
-```
-
-Test http 408 with curl:
-
-```bash
-curl --location -I --request GET 'http://localhost:3000/foo/3000' 
-```
-
-Use with custom error:
+### Use with custom error:
 
 ```go
 var ErrFooTimeOut = errors.New("foo context canceled")
@@ -117,7 +119,7 @@ func sleepWithContextWithCustomError(ctx context.Context, d time.Duration) error
 }
 ```
 
-Sample usage with a DB call:
+### Sample usage with a DB call:
 
 ```go
 func main() {

--- a/docs/middleware/timeout.md
+++ b/docs/middleware/timeout.md
@@ -86,7 +86,7 @@ curl -i http://localhost:3000/sleep/3000   # returns 408 Request Timeout
 | OnTimeout | `fiber.Handler`    | Handler executed when a timeout occurs. Defaults to returning `fiber.ErrRequestTimeout`. | `nil`  |
 | Errors    | `[]error`          | Custom errors treated as timeout errors.                            | `nil`  |
 
-### Use with custom error:
+### Use with custom error
 
 ```go
 var ErrFooTimeOut = errors.New("foo context canceled")
@@ -119,7 +119,7 @@ func sleepWithContextWithCustomError(ctx context.Context, d time.Duration) error
 }
 ```
 
-### Sample usage with a DB call:
+### Sample usage with a Database call
 
 ```go
 func main() {

--- a/docs/middleware/timeout.md
+++ b/docs/middleware/timeout.md
@@ -119,7 +119,7 @@ func sleepWithContextWithCustomError(ctx context.Context, d time.Duration) error
 }
 ```
 
-### Sample usage with a Database call
+### Sample usage with a database call
 
 ```go
 func main() {

